C_image_processing/filters/sepia_filter.c

```diff
@@ -17,58 +19,63 @@ int sepia_filter(const char *inputFile, const char *outputFile)
         return 1;
     }
 
-    // Read header info of image
-    unsigned char headerInfo[54];
-    for (int i = 0; i < 54; i++)
+    // read header info of image
+    for (i = 0; i < 54; i++)
     {
-        headerInfo[i] = getc(fileIn);
+        byte[i] = getc(fileIn);
     }
 
-    // Write header info to output file
-    fwrite(headerInfo, sizeof(unsigned char), 54, fileOut);
+    // write header info to output file
+    fwrite(byte, sizeof(unsigned char), 54, fileOut);
+    // extract attributes from image header
+    int height = *(int *)&byte[18];
+    int width = *(int *)&byte[22];
+    int bitDepth = *(int *)&byte[28];
+    int size = height * width;
 
-    // Extract.. of image from header info
-    int height = *(int *)&headerInfo[18];
-    int width = *(int *)&headerInfo[22];
-    int bitDepth = *(int *)&headerInfo[28];
-    int pixelsInImage = height * width;
+    // check if image has a color table
+    unsigned char buffer[size][3];
 
-    // Store image data in buffer
-    unsigned char(*buffer)[3] = malloc(pixelsInImage * sizeof(*buffer));
-    if (buffer == NULL)
```

Comment:
Why has a buffer Error check been removed, I think it would significantly improve the readability of the code while debugging, though might make a suggestion that we require an Error checking file, to lessen the clutter of the file.

Reply:
added error check back